src/linear_combination.jl

```diff
@@ -50,6 +49,15 @@ function apply(table, LC::LinearCombination; cols=_get_cols(table), header=nothi
     return Tables.materializer(table)(_to_table(result, header))
 end
 
+function apply_append(
+    A::AbstractArray{<:Real, N}, LC::LinearCombination; append_dim, kwargs...
+)::AbstractArray{<:Real, N} where N
+    # A was reduced along the append_dim so we must reshape the result setting that dim to 1
+    new_size = collect(size(A))
+    setindex!(new_size, 1, dim(A, append_dim))
```

Comment:
this was unfortunately necessary to introduce NamedDims as a dependency.

Without `NamedDims.dim` there was no way to map the dimname to the dim - and so no way to know how to reshape the output of LinearCombination so it could be `cat` to the input.

Reply:
This comes down to `setindex!` though, right? Because it has to use an integer index. Just wondering if there is an alternative outside of `setindex!`.

Reply:
>This comes down to setindex! though, right? Because it has to use an integer index.

yeah exactly. I'm not sure what alternative exists though. I don't think there's anything in base that assumes a dim can be a symbol.